lib/internal/test_runner/snapshot.js

```diff
@@ -147,6 +147,9 @@ class SnapshotManager {
       }
 
       this.snapshots = context.exports;
+      for (const [key, value] of Object.entries(this.snapshots)) {
```

Comment:
This should be replaced with a primordial implementation, see https://github.com/nodejs/node/blob/main/doc/contributing/primordials.md

Reply:
This will look something like this:

```js
// Add to the imports at the top of the file.
const {
  ObjectEntries,
} = primordials;

const entries = ObjectEntries(context.exports);

for (let i = 0; i < entries.length; i++) {
  const { 0: key, 1: value } = entries[i];
  this.snapshots[key] = templateEscape(value);
}
```

(Let's also update to operator on `context.exports` before assigning to `this.snapshots`.

Reply:
Out of curiosity: Can you explain why this shouldn't use array destructuring and for-of?

Also, what about something like this, which looks a bit more natural to me, and also avoids these, and even the need for the primordial:

```js
for (const key in context.exports) {
  this.snapshots[key] = templateEscape(context.exports[key]);
}
```

🤔

Reply:
I believe `for-in` loops are allowed, so that should be fine.

> Can you explain why this shouldn't use array destructuring and for-of?

The idea is that user code can change the way those things work and potentially break the test runner. There is an explanation of both of those in https://github.com/nodejs/node/blob/main/doc/contributing/primordials.md#implicit-use-of-user-mutable-methods.

Reply:
Oh sorry, I overlooked that section. 😅 Alright, thanks, I will make the change, then! 😀